posts/2024/2024_08_22_inigo_final_report.rst

```diff
@@ -0,0 +1,309 @@
+
+  * The `FINTA <https://doi.org/10.1016/j.media.2021.102126>`_ and `GESTA <https://doi.org/10.1016/j.media.2023.102761>`_ papers because they provide a relatively simple AE architecture with an open source sampling algorithm, easy to reuse.
+
+  * `Variational AutoEncoders for Regression <https://doi.org/10.1007/978-3-030-32245-8_91>`_, which provided a good starting point for conditional Variational AutoEncoders with direct application to brain aging, related to the project's objectives.
+
+  * `Attribute-based Regularization of latent spaces for variational AutoEncoders <https://doi.org/10.1007/s00521-020-05270-2>`_ and `Adversarial AutoEncoders <http://arxiv.org/abs/1511.05644>`_, which inspired the use of Attribute-based regularization for conditioning on a continuous variable (age); and the use of adversarial training for conditioning on categoric variables (fiber bundle or neurodegeneration status), respectively.
+
+* **Replicated the** `FINTA <https://doi.org/10.1016/j.media.2021.102126>`_ **architecture originally implemented in PyTorch (vanilla AutoEncoder, not variational) using TensorFlow2+Keras.** 
+  Validated the results using the FiberCup dataset. The model is found in the ``ae_model.py`` module of the `TractoEncoder GSoC <https://github.com/itellaetxe/tractoencoder_gsoc>`_ repository. The architecture can be summarized as in the following image:
+
+  .. image:: /_static/images/gsoc/2024/inigo/inigo_vanilla_autoencoder.png
+    :alt: AE architecture diagram.
+    :align: center
+    :width: 800
+
+  * Weight and bias initializers are different in PyTorch compared to TensorFlow2, so the PyTorch behavior was replicated using custom initializers. Different weight & bias initialization strategies can lead to drastically different training results, so this step took extra care.
+  * The upsampling layers of the Decoder block use linear interpolation in PyTorch by default, whereas in TensorFlow2, there is no native implementation for this, and nearest neighbor (NN) interpolation is used instead. This is a significant difference in implementations, and to replicate the PyTorch behavior a custom linear interpolating upsampling layer was implemented in TF2. However, after training with both NN and linear interpolation, the results were very similar, so the custom layer was not used in the final implementation. This work was developed in a `separate branch <https://github.com/itellaetxe/tractoencoder_gsoc/tree/feature_linear_upsampling>`_.
+  * Training was run for 120 epochs, using a data set containing plausible and implausible streamlines. All the training experiments in my GSoC work were done with this data set. The figure below shows a summary of the replication results, which consist of running a set of unseen plausible streamlines through the model (encoder and decoder):
+
+    .. image:: /_static/images/gsoc/2024/inigo/fibercup_replicated.png
```

Comment:
Also anything with caption/legend will be a figure not image.

```
.. figure:: /_static/images/gsoc/2024/inigo/fibercup_replicated.png
      :alt: Replication of the FINTA architecture results on the FiberCup dataset.
      :align: center
      :width: 600
      :class: custom-margin
  
      We see that the Keras result ...
 ```
 If it was not put as figure, the text below will not be rendered as caption.

Reply:
understood, ty, done

Reply:
It seems that with the proposed changes the caption does not render as expected:

![image](https://github.com/user-attachments/assets/2e082a6f-4348-4c65-aa4d-5d06f0fd1c35)

I believe I implemented your suggestion correctly, @deka27, but idk why it is not working.

Reply:
Checking.

Reply:
The caption text, should be right below :width

Previous
![image](https://github.com/user-attachments/assets/d963eb6d-29b6-4c6d-a5a3-d2459ef089b1)
Now
![image](https://github.com/user-attachments/assets/e48ad377-64a6-4ff8-84f7-dd46c2a8a339)

Reply:
Thank you again Kaustav :)))